Frameworks/jdi-core/src/main/java/com/epam/jdi/uitests/core/interfaces/base/IBaseElement.java

```diff
@@ -7,32 +7,89 @@
 
 import com.epam.jdi.tools.func.JFunc;
 import com.epam.jdi.tools.func.JFunc1;
+import com.epam.jdi.uitests.core.actions.base.ElementActions;
 import com.epam.jdi.uitests.core.annotations.JDIAction;
 import com.epam.jdi.uitests.core.templates.base.LinkedElements;
 
-import static com.epam.jdi.uitests.core.actions.base.ElementActions.*;
 import static com.epam.jdi.uitests.core.logger.LogLevels.DEBUG;
 
+/**
+ * Interface for base behaviours of elements
+ */
 public interface IBaseElement extends IHasParent, INamed, IVisible {
+    /**
+     * Set type name
+     *
+     * @param typeName Specify type name
+     */
     void setTypeName(String typeName);
-    default String printContext() { return printContext.execute(this); }
+
+    /**
+     * Print current context
+     *
+     * @return Current context as String
+     */
+    default String printContext() {
+        return ElementActions.printContext.execute(this);
+    }
+
+    /**
+     * Get engine
+     *
+     * @return Current engine
+     */
     IEngine engine();
+
+    /**
+     * Set timeout
+     *
+     * @param timoutSec Specify current timeout
+     * @return This element
+     */
     <T extends IBaseElement> T withTimeout(int timoutSec);
+
+    /**
+     * Set driver name
+     *
+     * @param driverName Specify current driver name
+     */
     void setDriverName(String driverName);
+
+    /**
+     * Get useCache parameter
+     *
+     * @return Does current engine uses caching or not
+     */
     boolean isUseCache();
+
+    /**
+     * Set useCache parameter
+     *
+     * @param useCache Specify using cache parameter for current engine
+     */
     void setUseCache(boolean useCache);
+
     /**
-     *  Wait condition
+     * Wait condition
      */
     @JDIAction(value = "Wait action", level = DEBUG)
-    default void wait(JFunc<Boolean> condition) { wait.execute(this, condition); }
+    default void wait(JFunc<Boolean> condition) {
+        ElementActions.wait.execute(this, condition);
+    }
+
     /**
-     *  Wait condition
+     * Wait condition
      */
     @JDIAction(value = "Wait result")
     default <R> R wait(JFunc<R> get, JFunc1<R, Boolean> condition) {
-        return (R) getResultWithCondition.execute(this, () -> get.execute(), r -> condition.execute((R)r));
```

Comment:
That's change as well

Reply:
I just specified class, which static method was called for readability